internal/collection/db.go

```diff
@@ -0,0 +1,69 @@
+// Package db is used for low-level operations related to communication with the database.
+package collection
+
+import (
+	"github.com/rs/zerolog"
+	"github.com/rs/zerolog/log"
+	"github.com/xujiajun/nutsdb"
+)
+
+// Repository interface describes the basic methods for obtaining and modifying data in a database.
+type Repository interface {
+	Save(ent map[string][]string) error
+	GetCollectionName() string
+}
+
+// NutsRepository is repository interface implementation for the NutsDB database.
+type NutsRepository struct {
+	db      *nutsdb.DB
+	colName string
+	l       zerolog.Logger
+}
+
+// Config describes the basic database configuration.
+type Config struct {
+	File string
+}
+
+// CollectionName sets the name for the collection to be contained in the repository.
+type Name string
+
+// NewNutRepo function-constructor to NutsRepository.
+func NewNutRepo(colName Name, db *nutsdb.DB) *NutsRepository {
+	l := log.With().Str("collection name", string(colName)).Logger()
+	l.Debug().Msg("initialize repository")
+	return &NutsRepository{db: db, colName: string(colName), l: l}
+}
+
+// GetCollectionName returns the name of the collection specified for this repository.
+func (nr *NutsRepository) GetCollectionName() string {
+	return nr.colName
+}
+
+// Save saves data to the collection of this repository.
+func (nr *NutsRepository) Save(ent map[string][]string) error {
+
+	nr.l.Debug().Interface("data", ent).Msg("start inserting data")
+
+	return nr.db.Update(func(tx *nutsdb.Tx) error {
+		for i := range ent {
+			vals := ent[i]
+			data := make([][]byte, 0, len(vals))
+			for j := range vals {
+				data = append(data, []byte(vals[j]))
+			}
+			if err := tx.SAdd(nr.colName, []byte(i), data...); err != nil {
```

Comment:
Если список коллекций с настройками потом будет храниться в отдельном бакете, то стоит добавить префикс к имени бакета. Чтоб потом не было коллизий и не нужно было делать ограничений на имена.
Но если предполагается, что коллекции и настройку будут храниться не в этой же БД, то ок.

Reply:
Согласен

Reply:
Я оставил коммент в коде по поводу этой идеи

Reply:
А мы же настройки коллекции можем хранить в самой коллекции в базе данных

Reply:
Там в одном bucket есть несколько структур хранения. И например если вы сохраняете данные с помощью SAdd (key, []values), то просто Get(key) ничего не даст насколько я помню

Reply:
> All keys in a bucket must be unique. Bucket can be interpreted as a table or namespace. So you can store the same key in different bucket.

> Also, this bucket is related to the data structure you use. Different data index structures that use the same bucket are also different. For example, you define a bucket named bucket_foo, so you need to use the list data structure, use tx.RPush to add data, you must query or retrieve from this bucket_foo data structure, use tx.RPop, tx.LRange, etc. You cannot use tx.Get (same index type as tx.GetAll, tx.Put, tx.Delete, tx.RangeScan, etc.) to read the data in this bucket_foo, because the index structure is different. Other data structures such as Set, Sorted Set are the same.

И я не нашёл метода получить все бакеты, поэтому их список всё таки потребуется где-то хранить.

Reply:
Согласен, нельзя получить все бакеты